packages/core/src/dispatcher.ts

```diff
@@ -389,6 +419,51 @@ export class Dispatcher {
     // TODO: Handle 'anchorService' if present in message
   }
 
+  _speedUpResponses() {
+    if (this.responseDelayMS - RESPONSE_DELAY_INCREMENT_MS > MINIMUM_RESPONSE_DELAY_MS) {
+        this.responseDelayMS -= RESPONSE_DELAY_INCREMENT_MS
+    } else {
+        this.responseDelayMS = MINIMUM_RESPONSE_DELAY_MS
+    }
+  }
+
+  _slowDownResponses() {
+    const newResponseDelayMS = (this.responseDelayMS + 1) * RESPONSE_DELAY_SLOWDOWN_FACTOR
+    if (newResponseDelayMS < MAXIMUM_RESPONSE_DELAY_MS) {
+        this.responseDelayMS = newResponseDelayMS
+    } else {
+        this.responseDelayMS = MAXIMUM_RESPONSE_DELAY_MS
+    }
+  }
+
+  _shouldAnswer (streamId, tip, streamState) {
+     // do we have the best answer for this?  was it already answered?
+
+     const answers = this.queryResponsesCache.get(streamId.toString())
+     // no one else answered, we should
+     if (answers == undefined) {
+         return true
+     }
+
+     // someone else already gave the same answer, we should not
+     if (answers.includes(tip)) {
+         return false
+     }
+
+     // ok we disagree, but whose answer is better?  now we have to look at the streamState
+
+     // if we know about every answer someone else gave, but we have a different answer
+     // then we know best
+     if (answers.every((answerTip) => streamState.log.includes(answerTip))) {
+         return true
+     }
+
+     // we didn't know about all the answers someone else gave so they probably know better
+     // note we have to prune attackers who are just saying junk, that should be separate thing
+     // the guy whose query it was will report them when he tries to anchor it and checks the sig
+     return false
```

Comment:
What if we have a conflicting answer. Ex A->B and A->C

Reply:
yeah we could add a case for this also